extensions/native/transpiler/src/lib.rs

```diff
@@ -0,0 +1,52 @@
+use openvm_instructions::{instruction::Instruction, riscv::RV32_REGISTER_NUM_LIMBS, VmOpcode};
+use openvm_native_serialization::{
+    GAP_INDICATOR, LONG_FORM_INSTRUCTION_INDICATOR, VARIABLE_REGISTER_INDICATOR,
+};
+use openvm_transpiler::{TranspilerExtension, TranspilerOutput};
+use p3_field::PrimeField32;
+
+pub struct LongFormTranspilerExtension;
+
+impl<F: PrimeField32> TranspilerExtension<F> for LongFormTranspilerExtension {
+    fn process_custom(&self, instruction_stream: &[u32]) -> Option<TranspilerOutput<F>> {
+        if instruction_stream[0] == LONG_FORM_INSTRUCTION_INDICATOR {
+            let num_operands = instruction_stream[1] as usize;
+            let opcode = VmOpcode::from_usize(instruction_stream[2] as usize);
+            let mut operands = vec![];
+            let mut j = 3;
+            for _ in 0..num_operands {
+                if instruction_stream[j] == VARIABLE_REGISTER_INDICATOR {
```

Comment:
what is the purpose of VARIABLE_REGISTER_INDICATOR? it seems weird to have here in this general processing function

Reply:
I somehow missed this doc :O